src/Util.hs

```diff
@@ -24,31 +26,42 @@ import           Data.Data
 
 import           Data.Aeson.Types
 
+import           Haskell
+
 newtype List a = List {fromList :: [a]}
   deriving (Eq, Show, Data, Typeable)
 
 instance FromJSON a => FromJSON (List a) where
   parseJSON v = List <$> (parseJSON v <|> (return <$> parseJSON v))
 
+type GhcOption = String
+
+parseMain :: String -> (FilePath, [GhcOption])
+parseMain main = case reverse name of
+  x : _ | isQualifiedIdentifier name && x `notElem` ["hs", "lhs"] -> (intercalate "/" (init name) ++ ".hs", ["-main-is " ++ main])
+  _ | isModule name -> (intercalate "/" name ++ ".hs", ["-main-is " ++ main])
+  _ -> (main, [])
+  where
+    name = splitOn '.' main
+
+splitOn :: Char -> String -> [String]
```

Comment:
This could have a more generic type: `a -> [a] -> [[a]]`. Don't know if that would be good.

There is `wordsBy` in the `split` package. Maybe you want to use the same name? Although, it takes a predicate as first argument.